lib/CodeGen/PagerandoBinning.cpp

```diff
@@ -47,16 +41,19 @@ class PagerandoBinning : public ModulePass {
   bool runOnModule(Module &M) override;
 
   void getAnalysisUsage(AnalysisUsage &AU) const override {
+    ModulePass::getAnalysisUsage(AU);
     AU.addRequired<MachineModuleInfo>();
     AU.addPreserved<MachineModuleInfo>();
     AU.setPreservesAll();
-    ModulePass::getAnalysisUsage(AU);
   }
 
 private:
-  // Map from free space -> Bin
-  typedef std::multimap<unsigned, Bin> BinMap;
-  BinMap Bins;
+  static constexpr unsigned DefaultBin = 0;
+  static constexpr unsigned BinSize = 4096;     // one page
+  static constexpr unsigned MinFreeSpace = 64;  // cache line (32 or 64 on ARM)
```

Comment:
If this mechanism is not useful, I can also remove it again.
My current assumption: functions can shrink (due to optimizing call sites to direct calls for intra-bin calls), but can't possibly grow?!